### PR TITLE
Gt/simple submarine diffuser

### PR DESCRIPTION
**Pull request by gregtucker**
Diffusion-based marine sediment transport model that uses a water-depth-dependent diffusivity, with:
- diffusivity above wave base is uniform (except within tidal range)
- diffusivity below wave base decreases exponentially with water depth
- if specified tidal range is nonzero, diffusivity varies smoothly across mean sea level using a tanh function (so it is about 88% of maximum at the mean low tide level, and 12% of maximum at the mean high tide level)
- above mean sea level (i.e., on land), apart from a small diffusivity associated with tidal effects, diffusivity is arbitrarily tiny (1e-20 m2/y), with this small positive value intended to avoid division by zero in the base class LinearDiffuser.

### CODE REVIEW

File: tests/components/marine_sediment_transport/test_simple_submarine_diffuser.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    assert_array_equal(df, [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 0.0, 1.0, 1.0, 1.0, 1.0, 1.0])
+
+    ssd = SimpleSubmarineDiffuser(grid, tidal_range=2.0)
+    df = ssd.depth_function(depth)
+    assert_array_almost_equal(
+        df, [0.5, 0.5, 0.5, 0.5, 0.5, 0.880797, 0.119203, 0.5, 0.5, 0.5, 0.5, 0.5]
+    )
+
+
+if __name__ == "__main__":
```
Comment (mcflugen) [inline tests/components/marine_sediment_transport/test_simple_submarine_diffuser.py:142]: Remove this block?

**Review by mcflugen — CHANGES_REQUESTED** [review]
Looks good @gregtucker! 👍 It looks like there is a stray XCode file that should be removed (*default.profraw*).

File: tests/components/marine_sediment_transport/test_simple_submarine_diffuser.py (hunk 1 above)
Comment (gregtucker) [reply tests/components/marine_sediment_transport/test_simple_submarine_diffuser.py]: done

Comment (gregtucker) [reply tests/components/marine_sediment_transport/test_simple_submarine_diffuser.py]: and stray default.profraw file removed

**Review by mcflugen — APPROVED** [review]

**Pull request merged**